core/ir/x86/decode_private.h

```diff
@@ -561,6 +561,14 @@ extern const instr_info_t xop_extensions[];
 extern const instr_info_t evex_prefix_extensions[][2];
 extern const instr_info_t evex_Wb_extensions[][4];
 
+/* A table that associates opcodes with corresponding categories.
+ * When a new opcode is added, it will be automatically assigned the default
+ * category of "DR_INSTR_CATEGORY_UNCATEGORIZED".
+ * It is important to update the table by assigning appropriate categories
+ * to these new opcodes.
+ */
+extern const uint category_by_opcode[OP_AFTER_LAST];
```

Comment:
The question is, should these instead be added as a new field in instr_info_t.  To decide: 
+ Do they vary by encoding template? If that happens a bunch it may be better in instr_info_t instead of lots of callouts to custom code to deduce.
+ Is it easier to maintain when the category is in the encoding template instead of in a separate array we need to update?  OTOH if many opcodes have multiple  encoding entries and they all have identical categories maybe it is easier with a separate out-of-line table.

Reply:
Sounds much better with field in  instr_info_t.
Added category member for x86 only. could be issue with others architecture here?
Thx, Kirill